src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -67,6 +67,113 @@ internal class FeatureManagementKeyValueAdapter : IKeyValueAdapter
                 keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}", false.ToString()));
             }
 
+            if (featureFlag.Variants != null)
+            {
+                for (int i = 0; i < featureFlag.Variants.Count; i++)
+                {
+                    FeatureVariant featureVariant = featureFlag.Variants[i];
+
+                    keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.Variants}:{i}:Name", featureVariant.Name));
+
+                    if (featureVariant.ConfigurationValue != null)
+                    {
+                        keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.Variants}:{i}:ConfigurationValue", featureVariant.ConfigurationValue));
+                    }
+
+                    if (featureVariant.ConfigurationReference != null)
+                    {
+                        keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.Variants}:{i}:ConfigurationReference", featureVariant.ConfigurationReference));
+                    }
+
+                    if (featureVariant.StatusOverride != null)
+                    {
+                        keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.Variants}:{i}:StatusOverride", featureVariant.StatusOverride));
+                    }
+                }
+            }
+
+            if (featureFlag.Allocation != null)
+            {
+                FeatureAllocation featureAllocation = featureFlag.Allocation;
+
+                if (featureAllocation.DefaultWhenDisabled != null)
+                {
+                    keyValues.Add(new KeyValuePair<string, string>($"{FeatureManagementConstants.SectionName}:{featureFlag.Id}:{FeatureManagementConstants.Allocation}:DefaultWhenDisabled", featureAllocation.DefaultWhenDisabled));
```

Comment:
How come some strings are literals rather than appearing in FeatureManagementConstants?

Reply:
I see how I was inconsistent with that, so I think I can just make them all constants to simplify things.